### DISCUSSION TITLE
tokio-fs: How can I get file position like Python's file.tell()?

### DISCUSSION THREAD

**Discussion by ariesdevil** [tokio-rs/tokio#2556] [General] [answered]
How can I get a file position like Python's file.tell()? Does the `seek` API can do same thing?

**Comment by Darksonn** [comment, marked as answer]
The `seek` method returns the position after seeking, so you can seek by zero like this:
```rust
use tokio::io::AsyncSeekExt;
use std::io::SeekFrom;

let pos = file.seek(SeekFrom::Current(0)).await?;
println!("Current position in file is {}.", pos);
```

**Reply by ariesdevil** [reply]
It seems `seek` op will cause a syscall?

**Reply by Darksonn** [reply]
Yes it does. Note that Python's `tell()` probably also causes a syscall. You can of course also manually keep track of the position with an integer.

**Reply by ariesdevil** [reply]
Got it, thx
